analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CheckArgumentException.CSharp9.cs

```diff
@@ -95,9 +95,25 @@ static void Something(string localArg)
 }
 
 // https://github.com/SonarSource/sonar-dotnet/issues/5094
-public record Data(string Code)
+public record Data(string Code, string Name)
 {
-    public string Code { get; } = Code ?? throw new ArgumentNullException(nameof(Code)); // Noncompliant, FP - Code is a record parameter
+    public string Code { get; } = Code ?? throw new ArgumentNullException(nameof(Code)); // Compliant - Code is a record parameter
+    public string GetName()
+    {
+        return Name ?? throw new ArgumentNullException(nameof(Name)); // Compliant - Name is a record parameter
+    }
+
+    public string GetNameLambda() => Name ?? throw new ArgumentNullException(nameof(Name)); // Compliant - Name is a record parameter
+}
+
+public class TestClass
+{
+    public static string ClassProperty { get; }
+
+    public record TestRecord()
+    {
+        public string RecordProperty { get; } = ClassProperty ?? throw new ArgumentNullException(nameof(ClassProperty)); // Noncompliant
```

Comment:
Could you please add a test case similar with this one where the throw is inside a class inside a record with parameters and the throw. Similar with this:
```
public record R(string Prop)
{
    public class TestClass
    {
        public static string ClassProperty { get; }

        public string RecordProperty { get; } = ClassProperty ?? throw new ArgumentNullException(nameof(Prop));
    }
}
```